### DISCUSSION TITLE
classes/class_streampeertcp

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#60] [User-contributed notes]
# classes/class_streampeertcp

Inherits: StreamPeer< RefCounted< Object A stream peer that handles TCP connections. Description: A stream peer that handles TCP connections. This object can be used to connect to TCP servers, or a...

https://docs.godotengine.org/en/stable/classes/class_streampeertcp.html

**Comment by Soup-64** [comment]
Note that `poll()` is a required call to make for `StreamPeerTCP` to function. Even if you don't use `get_status()`, puts and gets will fail with a generic error otherwise [since it's no longer called implicitly in godot 4.x](https://github.com/godotengine/godot/issues/74033#issuecomment-1484056801).

**Comment by SteveSmith16384** [comment]
I think one aspect the docs don't explain is which methods block, e.g. connect_to_host().  Does it block, or do we call it and then periodically check whether we're connected?  And if it does block, does that mean we need to use threads?
